yt_dlp/extractor/hse.py

```diff
@@ -0,0 +1,99 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+from time import mktime, strptime
+
+
+class HSEIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?hse\.de/dpl/(?:p/product|c/tv-shows)?/(?P<id>[0-9]+)'
+    _GEO_COUNTRIES = ['DE']
+    _TESTS = [{
+        'url': 'https://www.hse.de/dpl/c/tv-shows/505350',
+        'info_dict': {
+            'id': '505350',
+            'ext': 'mp4',
+            'title': 'Pfeffinger Mode & Accessoires',
+            'timestamp': 1638806400,
+            'upload_date': '20211206',
+            'channel': 'HSE24',
+            'uploader': 'Arina Pirayesh'
+        },
+        'params': {
+            'skip_download': True,
+        },
+    }, {
+        'url': 'https://www.hse.de/dpl/p/product/408630',
+        'info_dict': {
+            'id': '408630',
+            'ext': 'mp4',
+            'title': 'Hose im Ponte-Mix',
+            'timestamp': 1603058400,
+            'upload_date': '20201018',
+            'channel': 'HSE24',
+            'uploader': 'Judith Williams'
+        },
+        'params': {
+            'skip_download': True,
+        },
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        json_str = self._html_search_regex(r'window\.__REDUX_DATA__ = ({.*});?', webpage, 'json_str')
+        json_str = json_str.replace('\n', '')
+        json_data = self._parse_json(json_str, video_id)
+
+        formats = []
+        if 'tvShowPage' in json_data:  # /dpl/c/tv-shows
```

Comment:
since the extraction paths are entirely separate, better to split this into two extractors